cdap-api/src/main/java/co/cask/cdap/api/dataset/DatasetProperties.java

```diff
@@ -69,6 +74,17 @@ protected Builder() {
     }
 
     /**
+     * Sets description of the dataset
+     *
+     * @param description dataset description
+     * @return this builder object to allow chaining
+     */
+    public Builder setDescription(String description) {
+      this.properties.put(DESCRIPTION, description);
```

Comment:
If you simply use "description" as a property name, then that may clash with an actual property name used by the dataset.